server/src/iec62209_service/api.py

```diff
@@ -68,40 +73,69 @@ async def get_index(settings: ApplicationSettings = Depends(get_app_settings)):
     return html_content
 
 
-@router.post("/training-set-generation:create", status_code=status.HTTP_204_NO_CONTENT)
-async def create_training_set(body: TrainingTestGeneration):
-    print(body)
-
-
-@router.post("/training-set-generation:xport", response_class=FileResponse)
-async def xport_training_set():
-    some_file_path = "../my_model.json"
-    return some_file_path
-
-
-@router.get("/training-set-generation/data")
-async def get_training_data():
-    return
-
-
-@router.get("/training-set-generation/distribution", response_class=FileResponse)
-async def get_training_distribution():
-    some_file_path = "../my_model.json"
-    return some_file_path
-
-
-
-@router.get("/demo/{name}", response_model=Demo)
-async def demo(body: Demo, name: str, enabled: MyEnum = MyEnum.BAR):
-    return Demo(x=body.x, y=body.y + 3, z=body.x + 33)
-
-
-@router.post("/uploadfile/")
-async def create_upload_file(file: UploadFile):
-    return {"filename": file.filename}
-
-
-@router.post("/sample")
-async def generate_sample():
-    work = Work()
-    print(work)
+# Training set generation
+
+# for data storage
+class TrainingSetGeneration:
+    sample: dict = {"headings": [], "rows": []}
+
+
+@router.get("/training-set-generation:distribution", response_class=FileResponse)
+async def get_training_set_distribution() -> FileResponse:
+    response = FileResponse(dirname(realpath(__file__)) + "/../../testdata/mwl.png")
+    response.media_type = "image/png"
+    return response
+
+
+@router.get("/training-set-generation:data", response_class=PlainTextResponse)
+async def get_training_set_data() -> PlainTextResponse:
+    response = PlainTextResponse(str(TrainingSetGeneration.sample))
+    return response
+
+
+@router.post("/training-set-generation:{operation}", response_class=JSONResponse)
```

Comment:
There is no need for this. Let the router filter for you

Create two entrypoints instead of an if statement in the implementation